CDP4Orm/Dao/Revision/RevisionDao.cs

```diff
@@ -168,38 +185,42 @@ public void WriteRevision(NpgsqlTransaction transaction, string partition, Thing
         public IEnumerable<Thing> ReadRevision(NpgsqlTransaction transaction, string partition, Guid thingIid, int revisionFrom, int revisionTo)
         {
             var resolveInfos = this.ResolveDao.Read(transaction, partition, new[] { thingIid }).ToArray();
-            if (resolveInfos.Length != 1)
-            {
-                throw new InvalidOperationException(string.Format("Multiple entries were found for {0}", thingIid));
-            }
 
-            var resolveInfo = resolveInfos.Single();
+            if (resolveInfos.Length > 0)
+            {
+                if (resolveInfos.Length > 1)
```

Comment:
Condition greater than one should be outside condition greater than zero since is returning an error.